client/extensions/woocommerce/app/promotions/promotion-validations.js

```diff
@@ -16,7 +16,7 @@ import { isEmpty, isUndefined } from 'lodash';
  * @return { string } Returns a validation error, or undefined if none.
  */
 export function validateCouponCode( fieldName, promotion, currency, showEmpty ) {
-	if ( showEmpty && isEmpty( promotion.couponCode ) ) {
+	if ( showEmpty && isEmpty( promotion.couponCode.trim() ) ) {
```

Comment:
I'll have to check, but I think `promotion.couponCode` may evaluate to `undefined` initially for a new promotion with no edits. This code will probably have to take that into account.